starlite/app.py

```diff
@@ -158,7 +158,7 @@ class Starlite(Router):
 
     def __init__(
```

Comment:
As previously discusses in #1120, this method should be [overloaded](https://docs.python.org/3/library/typing.html#typing.overload) so type checkers will assert that it's only ever called with either an `AppConfig` instance, **or** keyword args, not both. This way we can also ensure that `route_handlers` is not `None` when using the kwarg style, which should be checked explicitly aside from that as well IMO.

Reply:
As `on-app-init` plugins can also provide routes, and routes can be registered after app creation, it is hard to definitively say that `route_handlers` would _always_ be required in the case that a config object isn't directly passed to the constructor.

I don't think that will be a very common case, and the workaround in that instance would be to pass an empty list like we have to in 1.x when instantiating an app without handlers. But it is worth considering if we need to be so strict on that `route_handlers` requirement.

Reply:
@provinzkraut @peterschutt I am currently trying to understand the overload decorator but unfortunately I was unable to make a signature which would make sure that __init__ is called with either AppConfig or route_handlers not both 

Best I could come up with is, I would greatly appreciate some guidance
```
    @overload
    def __init__(self, config: AppConfig) -> None:
        ...
    
    @overload
    def __init__(self, route_handlers: List["ControllerRouterHandler"]) -> None:
        ...

```

Reply:
I don't think overloads are the correct way to handle this, @starlite-api/maintainers . Since we are dealing here with Kwargs its going to be hard and a workaround will be pretty ugly. Instead I suggest the following, @wassafshahzad:

Create a static "factory" method called `from_config` on the Starlite class:

```python
class Starlite:
    def init(...):
       ...
     
    @static
    def from_config(config: "AppConfig") -> Starlite:
          return Starlite(**config.dict())
```

Reply:
Why not a `classmethod?

Reply:
sure, a class method is also fine

Reply:
Thanks for all the help guys